src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx

```diff
@@ -29,63 +33,75 @@ import willBlurTextInputOnTapOutsideFunc from '@libs/willBlurTextInputOnTapOutsi
 import ParticipantLocalTime from '@pages/home/report/ParticipantLocalTime';
 import ReportDropUI from '@pages/home/report/ReportDropUI';
 import ReportTypingIndicator from '@pages/home/report/ReportTypingIndicator';
-import reportPropTypes from '@pages/reportPropTypes';
 import * as EmojiPickerActions from '@userActions/EmojiPickerAction';
 import * as Report from '@userActions/Report';
 import * as User from '@userActions/User';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type * as OnyxTypes from '@src/types/onyx';
+import type * as OnyxCommon from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import AttachmentPickerWithMenuItems from './AttachmentPickerWithMenuItems';
 import ComposerWithSuggestions from './ComposerWithSuggestions';
 import SendButton from './SendButton';
 
-const propTypes = {
-    /** A method to call when the form is submitted */
-    onSubmit: PropTypes.func.isRequired,
+type ComposerRef = {
+    blur: () => void;
+    focus: (shouldDelay?: boolean) => void;
+    replaceSelectionWithText: (text: string, shouldAddTrailSpace: boolean) => void;
+    prepareCommentAndResetComposer: () => string;
+    isFocused: () => boolean;
+};
+
+type SuggestionsRef = {
+    resetSuggestions: () => void;
+    onSelectionChange?: (event: NativeSyntheticEvent<TextInputSelectionChangeEventData>) => void;
+    triggerHotkeyActions: (event: KeyboardEvent) => boolean | undefined;
+    updateShouldShowSuggestionMenuToFalse: (shouldShowSuggestionMenu?: boolean) => void;
+    setShouldBlockSuggestionCalc: (shouldBlock: boolean) => void;
+    getSuggestions: () => Mention[] | Emoji[];
+};
 
-    /** The ID of the report actions will be created for */
-    reportID: PropTypes.string.isRequired,
+type ReportActionComposeOnyxProps = {
+    /** The NVP describing a user's block status */
+    blockedFromConcierge: OnyxEntry<OnyxTypes.BlockedFromConcierge>;
 
-    /** The report currently being looked at */
-    report: reportPropTypes,
+    /** Whether the composer input should be shown */
+    shouldShowComposeInput: OnyxEntry<boolean>;
+};
 
-    /** Is composer full size */
-    isComposerFullSize: PropTypes.bool,
+type ReportActionComposeProps = ReportActionComposeOnyxProps &
```

Comment:
Same, maybe we can somehow reuse or pick props from `ComposerWithSuggestionsProps` ?